### PR TITLE
fix: adjust optionEatAll for click >= 8

### LINKED ISSUE #491: renderer schema error
this command:

```
neurodocker generate docker --json --base-image debian:buster-slim --pkg-manager apt --install git
```

fails with a schema error:

```
jsonschema.exceptions.ValidationError: {'name': 'install', 'kwds': {'pkgs': "('git',)", 'opts': None}} is not valid under any of the given schemas
```

and that's because the `pkgs` schema is expecting a list and getting a string of tuple.

### PR DESCRIPTION
**Pull request by satra**
closes #491 

this is due to change in behavior of click.

### CODE REVIEW

File: neurodocker/cli/minify/tests/test_minify.py [hunk 1]
```diff
@@ -8,9 +8,11 @@
 docker = pytest.importorskip("docker", reason="docker-py not found")
 
 
+@pytest.mark.skip(reason="ptrace no longer supported under docker")
 def test_minify():
     client = docker.from_env()
-    container = client.containers.run("python:3.9-slim", detach=True, tty=True)
+    container = client.containers.run("python:3.9-slim", detach=True, tty=True,
+                                      platform="Linux/amd64", privileged=True)
```
Comment (kaczmarj) [inline neurodocker/cli/minify/tests/test_minify.py:15]: i wonder if adding `cap_add=["SYS_PTRACE"]`  would fix this. not a priority for now but hopefully we can get this working again. though I wonder if `privileged=True` implies that.

**Comment by kaczmarj** [comment]
wow! tests are passing!

**Comment by satra** [comment]
i had to remove minify for now. reprozip uses ptrace which seems to have been disabled in docker.

File: neurodocker/cli/minify/tests/test_minify.py (hunk 1 above)
Comment (satra) [reply neurodocker/cli/minify/tests/test_minify.py:15]: i have tried all variants, none of them worked on my m1 mac.

Comment (satra) [reply neurodocker/cli/minify/tests/test_minify.py:15]: and from docker's docs it seems they may have fully disabled this.

**Comment by kaczmarj** [comment]
we can look into it in the future. `strace` works for me in a docker container on my debian testing machine. haven't tried reprozip but strace working suggests that ptrace is available.

```bash
$ uname -a
Linux dash 6.0.0-6-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.0.12-1 (2022-12-09) x86_64 GNU/Linux
$ docker run --rm -it debian:bookworm bash -c 'apt-get update -qq && apt-get install -yqq strace && strace date' 
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package libunwind8:amd64.
(Reading database ... 6009 files and directories currently installed.)
Preparing to unpack .../libunwind8_1.6.2-3_amd64.deb ...
Unpacking libunwind8:amd64 (1.6.2-3) ...
Selecting previously unselected package strace.
Preparing to unpack .../strace_5.10-1_amd64.deb ...
Unpacking strace (5.10-1) ...
Setting up libunwind8:amd64 (1.6.2-3) ...
Setting up strace (5.10-1) ...
Processing triggers for libc-bin (2.36-7) ...
execve("/usr/bin/date", ["date"], 0x7ffec2e872e0 /* 7 vars */) = 0
brk(NULL)                               = 0x55e3f20e1000
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fe064452000
access("/etc/ld.so.preload", R_OK)      = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
newfstatat(3, "", {st_mode=S_IFREG|0644, st_size=4782, ...}, AT_EMPTY_PATH) = 0
mmap(NULL, 4782, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7fe064450000
close(3)                                = 0
openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0Ps\2\0\0\0\0\0"..., 832) = 832
pread64(3, "\6\0\0\0\4\0\0\0@\0\0\0\0\0\0\0@\0\0\0\0\0\0\0@\0\0\0\0\0\0\0"..., 784, 64) = 784
newfstatat(3, "", {st_mode=S_IFREG|0755, st_size=1922136, ...}, AT_EMPTY_PATH) = 0
pread64(3, "\6\0\0\0\4\0\0\0@\0\0\0\0\0\0\0@\0\0\0\0\0\0\0@\0\0\0\0\0\0\0"..., 784, 64) = 784
mmap(NULL, 1970000, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fe06426f000
mmap(0x7fe064295000, 1396736, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x26000) = 0x7fe064295000
mmap(0x7fe0643ea000, 339968, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x17b000) = 0x7fe0643ea000
mmap(0x7fe06443d000, 24576, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1ce000) = 0x7fe06443d000
mmap(0x7fe064443000, 53072, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7fe064443000
close(3)                                = 0
mmap(NULL, 12288, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fe06426c000
arch_prctl(ARCH_SET_FS, 0x7fe06426c740) = 0
set_tid_address(0x7fe06426ca10)         = 286
set_robust_list(0x7fe06426ca20, 24)     = 0
rseq(0x7fe06426d060, 0x20, 0, 0x53053053) = 0
mprotect(0x7fe06443d000, 16384, PROT_READ) = 0
mprotect(0x55e3f0569000, 4096, PROT_READ) = 0
mprotect(0x7fe064484000, 8192, PROT_READ) = 0
prlimit64(0, RLIMIT_STACK, NULL, {rlim_cur=8192*1024, rlim_max=RLIM64_INFINITY}) = 0
munmap(0x7fe064450000, 4782)            = 0
getrandom("\x20\x60\xb3\xa9\x46\xd0\x49\x07", 8, GRND_NONBLOCK) = 8
brk(NULL)                               = 0x55e3f20e1000
brk(0x55e3f2102000)                     = 0x55e3f2102000
openat(AT_FDCWD, "/etc/localtime", O_RDONLY|O_CLOEXEC) = 3
newfstatat(3, "", {st_mode=S_IFREG|0644, st_size=114, ...}, AT_EMPTY_PATH) = 0
newfstatat(3, "", {st_mode=S_IFREG|0644, st_size=114, ...}, AT_EMPTY_PATH) = 0
read(3, "TZif2\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 4096) = 114
lseek(3, -60, SEEK_CUR)                 = 54
read(3, "TZif2\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 4096) = 60
close(3)                                = 0
newfstatat(1, "", {st_mode=S_IFCHR|0620, st_rdev=makedev(0x88, 0), ...}, AT_EMPTY_PATH) = 0
write(1, "Sat Jan 14 21:53:28 UTC 2023\n", 29Sat Jan 14 21:53:28 UTC 2023
) = 29
close(1)                                = 0
close(2)                                = 0
exit_group(0)                           = ?
+++ exited with 0 +++
```

**Pull request merged**